src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    // Who invented this task.. i want to cry
+    const copyThis = [...this];
```

Comment:
Checklist#2 suggests to avoid creating needless copies of arrays and iterate existing array

Reply:
agree, you don't need to copy the array here